src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -175,6 +176,55 @@ MultiClientConnectStart(const char *nodeName, uint32 nodePort, const char *nodeD
 }
 
 
+/*
+ * MultiClientPlacementConnectStart asynchronously tries to establish a connection
+ * for a particular set of shard placements. If it succeeds, it returns the
+ * the connection id. Otherwise, it reports connectio error and returns
+ * INVALID_CONNECTION_ID.
+ */
+int32
+MultiClientPlacementConnectStart(List *placementAccessList, const char *userName)
+{
+	MultiConnection *connection = NULL;
+	ConnStatusType connStatusType = CONNECTION_OK;
+	int32 connectionId = AllocateConnectionId();
+	int connectionFlags = CONNECTION_PER_PLACEMENT; /* no cached connections for now */
```

Comment:
We probably don't want to set the `CONNECTION_PER_PLACEMENT` if we're not yet in a coordinated transaction. Otherwise it would prevent connection reuse.